java/ql/test/utils/modelgenerator/dataflow/p/Sinks.java

```diff
@@ -25,6 +25,14 @@ public void sink2(Object o) {}
   // neutral=p;Sinks;nosink;(Object);summary;df-generated
   public void nosink(Object o) {}
 
+  // Sink and Source defined in the extensible file next to the sink test.
+  // sink=p;Sinks;true;saveAndGet;(Object);;Argument[0];test-sink;df-generated
+  // neutral=p;Sinks;saveAndGet;(Object);summary;df-generated
```

Comment:
Can I just check that I understand: is the reason that we don't expect a source here because the df-generator doesn't need to generate a source because there is already a manual source?

Reply:
This is starting to become a bit convoluted and complicated:
It is only the `CaptureSinkModels.ext.yml` file that is extended with models, which means that the models in this file doesn't affect the `CaptureSourceModels` test, but only the `CaptureSinkModels` test. The `CaptureSouceModels` test doesn't generate a source, since the method is not a source (it returns null).

Reply:
Okay, that makes sense. I agree it's a bit complicated. The only thing I can think of to reduce the complexity would be to give all the tests the same models.

Follow-up question: Why are you adding a source model for `saveAndGet` to the CaptureSinkModels test?

Reply:
Because the issue that is being fixed in this PR is the missing sink generation in case there exist a `manual` source for the same callable (actually the case is a bit more subtle: The sink is not generated if there a `manual` source and `df-generated` sink. In production this would case a wobble where the sink will be generated on the first model generation run and then removed again in the subsequent model generation run).